fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/UpgradeOnboarding.kt

```diff
@@ -62,200 +50,96 @@ private enum class OnboardingState {
+                    UpgradeOnboardingState.Welcome -> OnboardingMetrics.welcomeCloseClicked.record(NoExtras())
+                    UpgradeOnboardingState.SyncSignIn -> OnboardingMetrics.syncCloseClicked.record(NoExtras())
+                }
+                onDismiss()
+            },
+            onPrimaryButtonClick = {
+                when (onboardingState) {
+                    UpgradeOnboardingState.Welcome -> {
+                        OnboardingMetrics.welcomeGetStartedClicked.record(NoExtras())
+                        if (isSyncSignIn) {
+                            onDismiss()
+                        } else {
+                            onboardingState = UpgradeOnboardingState.SyncSignIn
+                        }
+                    }
+                    UpgradeOnboardingState.SyncSignIn -> {
+                        OnboardingMetrics.syncSignInClicked.record(NoExtras())
+                        onSignInButtonClick()
+                    }
+                }
+            },
+            onSecondaryButtonClick = {
+                when (onboardingState) {
+                    UpgradeOnboardingState.Welcome -> {
+                        // Welcome does not have a secondary button.
+                    }
+                    UpgradeOnboardingState.SyncSignIn -> {
+                        OnboardingMetrics.syncSkipClicked.record(NoExtras())
                         onDismiss()
-                    } else {
-                        onboardingState = OnboardingState.SyncSignIn
                     }
-                },
-            )
-
-            OnboardingMetrics.welcomeCardImpression.record(NoExtras())
-        } else if (onboardingState == OnboardingState.SyncSignIn) {
-            OnboardingSyncSignInContent()
-
-            OnboardingSyncSignInBottomContent(
-                onboardingState = onboardingState,
-                onSignInButtonClick = {
-                    OnboardingMetrics.syncSignInClicked.record(NoExtras())
-                    onSignInButtonClick()
-                },
-                onSkipButtonClick = {
-                    OnboardingMetrics.syncSkipClicked.record(NoExtras())
-                    onDismiss()
-                },
-            )
-
-            OnboardingMetrics.syncCardImpression.record(NoExtras())
-        }
-    }
-}
-
-@Composable
-private fun OnboardingWelcomeBottomContent(
-    onboardingState: OnboardingState,
-    isSyncSignIn: Boolean,
-    onGetStartedButtonClick: () -> Unit,
-) {
-    Column(modifier = Modifier.padding(horizontal = 16.dp)) {
-        PrimaryButton(
-            text = stringResource(id = R.string.onboarding_home_get_started_button),
-            onClick = onGetStartedButtonClick,
+                }
+            },
+            modifier = Modifier.weight(1f),
         )
 
-        Spacer(modifier = Modifier.height(32.dp))
-
-        if (isSyncSignIn) {
-            Spacer(modifier = Modifier.height(6.dp))
-        } else {
+        if (!isSyncSignIn) {
```

Comment:
Changes look good to me. However, we have removed the following condition:

```
if (isSyncSignIn) {
            Spacer(modifier = Modifier.height(6.dp))
}
```
This simply fills in the 6dp height if the Indicators wasn't visible. While this probably doesn't amount to much in the grand scheme of things, but I rather ensure we are keeping the bottom padding of the Button to the bottom of the screen to still align at a total of 70dp. 

https://www.figma.com/file/d7l1iGabQYqnLcnQbHDGzM/Wallpaper-2.0-%2B-Onboarding-v106?node-id=6058%3A209483&t=4jqlRpeR0Cday80G-0

Reply:
good catch! I've updated this 👍